### PR TITLE
`cli` formatting of prompt to install missing package

### LINKED ISSUE #1060: Formatting of prompt to install missing package
I tried using a step that needed a package I did not have installed. It told me so which was nice but the formatting could use a line break.

``` r
library(recipes)
#> Loading required package: dplyr
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
#> 
#> Attaching package: 'recipes'
#> The following object is masked from 'package:stats':
#> 
#>     step
data(ames, package = "modeldata")

ames_rec <-
  recipe(Sale_Price ~ ., data = ames) %>%
  step_spline_natural(Longitude, deg_free = 2) %>%
  step_spline_natural(Latitude, deg_free = 2)
#> 1 package is needed for this step and is not installed. (splines2). Start a clean R session then run: install.packages("splines2")1 package is needed for this step and is not installed. (splines2). Start a clean R session then run: install.packages("splines2")
```

<sup>Created on 2022-11-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by hfrick**
closes #1060

``` r
library(recipes)
#> Loading required package: dplyr
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
#> 
#> Attaching package: 'recipes'
#> The following object is masked from 'package:stats':
#> 
#>     step
data(ames, package = "modeldata")

ames_rec <-
  recipe(Sale_Price ~ ., data = ames) %>%
  step_spline_natural(Longitude, deg_free = 2) %>%
  step_spline_natural(Latitude, deg_free = 2)
#> 1 package (splines2) is needed for this step but is not installed.
#> Start a clean R session then run: install.packages("splines2")
#> 1 package (splines2) is needed for this step but is not installed.
#> Start a clean R session then run: install.packages("splines2")
```

<sup>Created on 2022-11-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

**Review by EmilHvitfeldt — CHANGES_REQUESTED** [review]
I love it! great to see the the move to {cli}. Would you consider using `{.run }` to make the `install.packages(c("missing_pkg_1", "missing_pkg_2"))` clickable? I think that would be neat

https://cli.r-lib.org/reference/links.html?q=.run#click-to-run-code

File: R/pkg_check.R [hunk 1]
```diff
@@ -22,21 +22,14 @@ recipes_pkg_check <- function(pkg = NULL, ...) {
...
-      "Start a clean R session then run: "
-    )
-    cat(msg)
-
     install_opt <- quos(...)
     install_pkg <- pkg[!good]
+    n_install_pkgs <- sum(!good)
+
+    cli::cli_text(
+      "{n_install_pkgs} package{?s} ({.pkg {install_pkg}}) {?is/are} needed for this step but {?is/are} not installed."
```
Comment (EmilHvitfeldt) [inline R/pkg_check.R:30]:
```suggestion
      "{n_install_pkgs} package{?s} ({.pkg {install_pkg}}) {?is/are} needed ",
      "for this step but {?is/are} not installed."
```

we can split up this string to keep it inside 80 chars

**Comment by hfrick** [comment]
Oh yes, that is neat! We recommend starting a clean R session though so that conflicts a little with "just click here". Is the new session necessary? If not, I'd drop that and say "To install, run `install.packages("missing_pkg")`".

```
recipes_pkg_check(pkg = "missing_pkg", dependencies = NA)
1 package (missing_pkg) is needed for this step but is not installed.
Start a clean R session then run: install.packages("missing_pkg", dependencies = NA)
```

**Comment by hfrick** [comment]
I assume we don't need a new session for the install (maybe after) so I've updated to a simple `To install run: `

**Review by EmilHvitfeldt — APPROVED** [review]
Love it!

**Pull request merged**